components/formats-gpl/src/loci/formats/in/ND2Handler.java

```diff
@@ -690,25 +688,20 @@ else if (key.equals("sObjective") || key.equals("wsObjectiveName") ||
         if (magIndex >= 0) {
           String m =
             tokens[magIndex].substring(0, tokens[magIndex].indexOf('x'));
-          m = DataTools.sanitizeDouble(m);
-          if (m.length() > 0) {
-            mag = new Double(m);
-          }
+          mag = DataTools.parseDouble(m);
         }
         if (magIndex + 1 < tokens.length) immersion = tokens[magIndex + 1];
       }
       else if (key.endsWith("dTimeMSec")) {
-        long v = (long) Double.parseDouble(DataTools.sanitizeDouble(value));
-        if (!ts.contains(new Long(v))) {
-          ts.add(new Long(v));
+        Long v = DataTools.parseLong(value);;
+        if (!ts.contains(v)) {
```

Comment:
I wonder if `ts` should have been `LinkedHashSet` or somesuch, outwith this PR though.